admin-dev/themes/new-theme/js/components/form/form-object-mapper.js

```diff
@@ -0,0 +1,282 @@
+  }
+
+  /**
+   * Returns the input associated to a model field (in case the field is mapped to
+   * several inputs the default one, first configured, is always used).
+   *
+   * @param modelKey {string}
+   * @returns {undefined|jQuery}
+   */
+  getInput(modelKey) {
+    if (!Object.prototype.hasOwnProperty.call(this.modelMapping, modelKey)) {
+      return undefined;
+    }
+
+    return $(`[name="${this.modelMapping[modelKey]}"]`);
+  }
+
+  /**
+   * Set a value to a field of the object based on the model key, the object itself is updated
+   * of course but the mapped inputs are also synced (all of them if multiple). Events are also
+   * triggered to indicate the object has been updated (the general and the individual field ones).
+   *
+   * @param modelKey {string}
+   * @param value {*|{}}
+   */
+  set(modelKey, value) {
+    if (!Object.prototype.hasOwnProperty.call(this.modelMapping, modelKey) || value === this.get(modelKey)) {
+      return;
+    }
+
+    // First update the inputs then the model, so that the event is sent at last
+    this.updateInputValue(modelKey, value);
+    this.updateObjectByKey(modelKey, value);
+    this.eventEmitter.emit(this.modelUpdatedEventName, this.object);
+  }
+
+  /**
+   * Watches if changes happens from the form or via an event.
+   */
+  watchUpdates() {
+    this.$form.on('keyup change dp.change', ':input', _.debounce(
+      (event) => this.inputUpdated(event),
+      200,
+      {maxWait: 1000},
+    ));
+    this.eventEmitter.on(this.updateModelEventName, () => this.updateFullObject());
+  }
+
+  /**
+   * Triggered when a form input has been changed.
+   *
+   * @param event {Object}
+   */
+  inputUpdated(event) {
+    const target = event.currentTarget;
+
+    if (!Object.prototype.hasOwnProperty.call(this.formMapping, target.name)) {
+      return;
+    }
+
+    const updatedValue = $(target).val();
+    const updatedModelKey = this.formMapping[target.name];
+
+    // Update the mapped input fields
+    this.updateInputValue(updatedModelKey, updatedValue);
+
+    // Then update model and emit event
+    this.updateObjectByKey(updatedModelKey, updatedValue);
+    this.eventEmitter.emit(this.modelUpdatedEventName, this.object);
+  }
+
+  /**
+   * Update all the inputs mapped to a model key
+   *
+   * @param modelKey {string}
+   * @param value {*|{}}
+   */
+  updateInputValue(modelKey, value) {
+    const modelInputs = this.fullModelMapping[modelKey];
+
+    // Update linked inputs (when there is more than one input associated to the model field)
+    if (Array.isArray(modelInputs)) {
+      modelInputs.forEach((inputName) => {
+        this.updateInputByName(inputName, value);
+      });
+    } else {
+      this.updateInputByName(modelInputs, value);
+    }
+  }
+
+  /**
+   * Update individual input based on its name
+   *
+   * @param inputName {string}
+   * @param value {*|{}}
+   */
+  updateInputByName(inputName, value) {
+    const $input = $(`[name="${inputName}"]`);
```

Comment:
I think it could cause problems there.

If you have multiple forms on the same page, you've a chance about having the inputName multiple times (for example if you've the FR form, at the same time than the EN form but only hidden by a display: none; the field would probably have the same name isn't it?

Maybe you could select it with the main current form like this:

```suggestion
    const $input = $(`[name="${inputName}"]`, this.$form);
```

Or better, can we use IDs ?

Reply:
Multilang inputs are identified via their langId, so you would have `names[1]` `names[3]` and so on, so for this case we should be fine But I agree it's safer to precise the form as second parameter in case multiple forms share some common namings